templates/templates/.github/workflows/fogg_ci.yml.tmpl

```diff
@@ -1,86 +1,102 @@
+          app_id: {{`${{ secrets.CZI_GITHUB_HELPER_APP_ID }}`}}
+          private_key: {{`${{ secrets.CZI_GITHUB_HELPER_PK }}`}}
+      - uses: actions/checkout@v3
         with:
-          ssh-private-key: |
-          {{ range $idx, $secret := $githubActionsCI.SSHKeySecrets }}
-            {{`${{`}} secrets.{{ $secret }} {{`}}`}}
-          {{ end }}
-      {{ end }}
-      - run: FOGG_VERSION="v{{ $githubActionsCI.FoggVersion }}" make setup
-      - run: |
-          .fogg/bin/fogg apply
-          if [ "$(git status --porcelain)" != "" ]; then
-            echo "git tree is dirty after running fogg apply"
-            echo "ensure you run fogg apply on your branch"
-            git status
-            exit 1
-          fi
-{{- if not (eq (len $githubActionsCI.AWSProfiles) 0) }}
-      - run: aws configure set aws_access_key_id     {{`${{`}} secrets.IDACCT_AWS_ACCESS_KEY_ID {{`}}`}}     --profile _idacct
-      - run: aws configure set aws_secret_access_key {{`${{`}} secrets.IDACCT_AWS_SECRET_ACCESS_KEY {{`}}`}} --profile _idacct
-      - run: aws --profile _idacct sts get-caller-identity
-{{- end }}
-{{- range $profileName, $profile := $githubActionsCI.AWSProfiles }}
-      - run: aws configure set profile.{{ $profileName }}.role_arn arn:aws:iam::{{ $profile.AccountID }}:role/{{ $profile.RoleName}}
-      - run: aws configure set profile.{{ $profileName }}.source_profile _idacct
-      - run: aws --profile {{ $profileName }} sts get-caller-identity
-{{- end }}
-      # we only run the following if there are changes in the terraform/* directory
-      # to optimize CI time
-      # can use this filter to decide whether or not to run linters or tests.
-      - uses: dorny/paths-filter@v2.2.1
+          token: {{`${{ steps.generate_token.outputs.token }}`}}
+      - name: Cache Fogg
+        id: cache-fogg
+        uses: actions/cache@v3
+        with:
+          path: ~/.fogg/cache
+          key: fogg-cache-{{`${{ hashFiles('**/.fogg-version') }}`}}
+      - run: make setup
+      - run: .fogg/bin/fogg apply
+        env:
+          FOGG_GITHUBAPPTOKEN: {{`${{ steps.generate_token.outputs.token }}`}}
+      - uses: EndBug/add-and-commit@v9
+        with:
+          add: -A
+          message: |
+            commit from fogg_ci -- ran fogg apply and pushed
+  find-changed-dirs:
+    runs-on: ubuntu-latest
+    outputs:
+      allChanges: {{`${{ steps.changedDirs.outputs.allChanges }}`}}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2.10.2
         id: filter
         with:
+          initial-fetch-depth: '1'
+          list-files: json
           filters: |
-            terraform:
-              - 'terraform/**'
-  {{- range $component := $testBucket }}
-      - name: {{ $component.Dir }}
-        if: {{ "${{ steps.filter.outputs.terraform == 'true' }}" }}
+            changed:
+              - added|modified: 'terraform/**'
+      - uses: actions/github-script@v6
+        id: changedDirs
```

Comment:
take the changed files and find the directories associated with their env/account/module folder